packages/core/src/common/keybinding.ts

```diff
@@ -140,7 +150,22 @@ export class KeybindingRegistry {
 
         const commands = this.commands[commandId] || [];
         commands.push(binding);
-        this.commands[commandId] = bindings;
+        this.commands[commandId] = commands;
+    }
+
+    unregisterKeybindings(bindingsToRemove: Keybinding[]) {
+        for (const bindingToRemove of bindingsToRemove) {
+            const { keyCode, commandId } = bindingToRemove;
+            let bindings = this.keybindings[keyCode.keystroke] || [];
+
+            bindings = bindings.filter(binding => binding !== bindingToRemove);
```

Comment:
We should have a proper value comparison logic instead of relying on object identity.